vxquery-core/src/main/java/org/apache/vxquery/index/IndexAttributes.java

```diff
@@ -32,6 +32,12 @@
 
     public IndexAttributes(List<String> n, List<String> v, List<String> u, List<String> l, List<String> t,
             List<String> q) {
+        n.clear();
```

Comment:
Almost ... move these clear statements to a reset method. Initial the arrays in the constructor. Then the object will no longer have any arguments to the constructor.